### ISSUE TITLE
ng-bootstrap causes AOT to fail with module not found errors.

### ISSUE THREAD

**Issue by qdouble** [ng-bootstrap/ng-bootstrap#796]
### Bug description:
When I import @ng-bootstrap and run AOT, I get the following errors and more

```
ERROR in ./src/app/app.module.ngfactory.ts                                                                    
Module not found: Error: Can't resolve '../../node_modules/@ng-bootstrap/ng-bootstrap/alert/alert.ngfactory' i
n '/Users/qdouble/Sites/EXAMPLE_APPS/angular2webpack2-starter/src/app'                                        
 @ ./src/app/app.module.ngfactory.ts 84:15-93                                                                 
 @ ./src/main.browser.aot.ts                                                                                  
                                                                                                              
ERROR in ./src/app/app.module.ngfactory.ts                                                                    
Module not found: Error: Can't resolve '../../node_modules/@ng-bootstrap/ng-bootstrap/datepicker/datepicker.ng
factory' in '/Users/qdouble/Sites/EXAMPLE_APPS/angular2webpack2-starter/src/app'                              
 @ ./src/app/app.module.ngfactory.ts 85:15-103                                                                
 @ ./src/main.browser.aot.ts                                                                                  
                                                                                                              
ERROR in ./src/app/app.module.ngfactory.ts                                                                    
Module not found: Error: Can't resolve '../../node_modules/@ng-bootstrap/ng-bootstrap/popover/popover.ngfactor
y' in '/Users/qdouble/Sites/EXAMPLE_APPS/angular2webpack2-starter/src/app'                                    
 @ ./src/app/app.module.ngfactory.ts 88:15-97                                                                 
 @ ./src/main.browser.aot.ts    
```

When I check the compiled files, it appears that many components are not being generated. 

### Version of Angular, ng-bootstrap, and Bootstrap:

Angular: 2.0.1

ng-bootstrap: 1.0.0-alpha.6

**Comment by pkozlowski-opensource** [comment]
We need to investigate and might need to re-export all directives from `index.ts` (which would be less than ideal...). My understanding is the at the whole AOT + libs story has some gaps / moving parts so we will need several adjustments before things are seamless.

**Comment by pkozlowski-opensource** [comment]
A (possibly) useful ref: https://github.com/angular/angular/issues/11689

**Comment by qdouble** [comment]
@pkozlowski-opensource I already have outdir and gendir that the person suggested in the link... I don't think there is anything wrong in particular settings, as I have material design branch that works just fine...I was trying to add a bootstrap branch to my seed starter project which AOT works otherwise fine with, but this is blocking creating bootstrap branch: https://github.com/qdouble/angular2webpack2-starter

**Comment by pkozlowski-opensource** [comment]
The most probable suspect: https://github.com/angular/angular/issues/11889

We need to investigate by creating a minimal project that uses ng-bootstrap.

**Comment by qdouble** [comment]
@pkozlowski-opensource is there something blocking this being fixed or do you need me to create a more minimal repo first?

**Comment by qdouble** [comment]
@pkozlowski-opensource I just uploaded the bootstrap branch of the starter project that I've been working on: https://github.com/qdouble/angular2webpack2-starter/tree/boostrap 

To reproduce the error, after installing. just run `npm run compile`

**Comment by pkozlowski-opensource** [comment]
@qdouble thnx for the repro. There are still a bunch of errors in the AoT on the ng2 side and I want to take time in the coming days to fix them upstream or work-around here (but honestly I would rather prefer fixing stuff in the Angular itself as it will benefit the entire community).

**Comment by pkozlowski-opensource** [comment]
@qdouble thnx to your repro I could just confirm that we are bumping into https://github.com/angular/angular/issues/11889 (or a slight variation of it). I need to do a bit more testing but I will either check with Tobias to see how far the fix is or will just work-around it for now as we were planning to export things anyway in #779

Thnx for bearing with us, a fix is not far away :-)

**Comment by smith64fx** [comment]
I hope this get fixed soon. ng-bootstrap is the only thing in our projects that's preventing AoT/Universal. If I just knew, where to look at then I could help you guys 🙃

**Comment by pkozlowski-opensource** [comment]
@smith64fx AoT is a top priority for us, all hands on the deck :-) I'm pretty positive that we are going to have a fix / workaround by Mon.

**Comment by smith64fx** [comment]
![image](https://cloud.githubusercontent.com/assets/3246782/19211578/056be436-8d40-11e6-8769-161c21615321.png)

Yeah I think so, too. Nothing is impossible!

**Comment by pkozlowski-opensource** [comment]
Ha! We've got a fix in #854. It just need review. I'm going to land it and cut a new release as soon as it is reviewed.

**Comment by pkozlowski-opensource** [comment]
For anyone tracking AoT compatibility - I've just pushed alpha.7 to npm so you can download it and take for the test drive. Be sure to check the [breaking changes section](https://github.com/ng-bootstrap/ng-bootstrap/blob/master/CHANGELOG.md#breaking-changes) while updating! The demo site updated to alpha.7 should be up shortly.
